src/DynamoCoreWpf/Utilities/CrashReportTool.cs

```diff
@@ -199,30 +199,59 @@ internal static bool ShowCrashErrorReportWindow(DynamoViewModel viewModel, Crash
                 {
                     string logFile = Path.Combine(cerDir.FullName, "DynamoLog.log");
 
-                    File.Copy(model.Logger.LogPath, logFile);
-                    // might be usefull to dump all loaded Packages into
-                    // the log at this point.
-                    filesToSend.Add(logFile);
+                    try
+                    {
+                        File.Copy(model.Logger.LogPath, logFile);
+                        // might be useful to dump all loaded Packages into
+                        // the log at this point.
+                        filesToSend.Add(logFile);
+                    }
+                    catch (Exception ex)
+                    {
+                        model?.Logger?.LogError($"Failed to add DynamoLog.log to CER with the following error : {ex.Message}");
+                    }
                 }
 
                 if (args.SendSettingsFile && model != null)
                 {
                     string settingsFile = Path.Combine(cerDir.FullName, "DynamoSettings.xml");
-                    File.Copy(model.PathManager.PreferenceFilePath, settingsFile);
 
-                    filesToSend.Add(settingsFile);
+                    try
+                    {
```

Comment:
If these try-catches are to prevent crashing during the absence of any of these files (like the settings file), is it not better to add checks if the files exist instead?

Reply:
Yes, try-catch is maybe overkill here.